services/volumesrv/store.go

```diff
@@ -11,21 +17,85 @@ func NewPostgresStore() Store {
 }
 
 func (p postgresStore) VolumeAdd(projectID string, volume types.Volume) error {
-	//TODO implement me
-	panic("implement me")
+	ctx := context.Background()
+	_, err := db.Q.VolumeCreate(ctx, db.VolumeCreateParams{
+		ID:        volume.ID,
+		ProjectID: projectID,
+		Provider:  volume.Provider.String(),
+	})
+	if err != nil {
+		return &types.Error{
+			Code:    http.StatusBadRequest,
+			Message: "Volume could not be created",
+			Err:     err,
+		}
+	}
+
+	return nil
+}
+
+func (p postgresStore) VolumeList(projectID string) ([]types.Volume, error) {
+	vols, err := db.Q.VolumeList(context.Background(), projectID)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
+		}
```

Comment:
Why is this nil! This should really not makes its way into a PR.

Reply:
this is a pretty normal way of handling returning an empty list

Reply:
Ok this i my bad! You're right that we probably don't want to return an error on a no volumes.